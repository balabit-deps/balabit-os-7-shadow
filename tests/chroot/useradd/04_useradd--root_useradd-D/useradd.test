#!/bin/sh

set -e

cd $(dirname $0)

. ../../../common/config.sh
. ../../../common/log.sh

log_start "$0" "useradd can list defaults from a chroot"

save_config

# restore the files on exit
trap 'log_status "$0" "FAILURE"; clean_chroot; restore_config' 0

change_config

prepare_chroot

echo -n "List defaults in chroot (useradd --root $PWD/tmp/root foo)..."
useradd -D --root $PWD/tmp/root > tmp/useradd.out
echo "OK"

echo "useradd reported:"
echo "======================================================================="
cat tmp/useradd.out
echo "======================================================================="
echo -n "Check the usage message..."
diff -au data/useradd.out tmp/useradd.out
echo "OK."
rm -f tmp/useradd.out

echo -n "Check the passwd file..."
../../../common/compare_file.pl config/etc/passwd /etc/passwd
../../../common/compare_file.pl config_chroot/etc/passwd tmp/root/etc/passwd
echo "OK"
echo -n "Check the group file..."
../../../common/compare_file.pl config/etc/group /etc/group
../../../common/compare_file.pl config_chroot/etc//group tmp/root/etc/group
echo "OK"
echo -n "Check the shadow file..."
../../../common/compare_file.pl config/etc/shadow /etc/shadow
../../../common/compare_file.pl config_chroot/etc/shadow tmp/root/etc/shadow
echo "OK"
echo -n "Check the gshadow file..."
../../../common/compare_file.pl config/etc/gshadow /etc/gshadow
../../../common/compare_file.pl config_chroot/etc/gshadow tmp/root/etc/gshadow
echo "OK"

rm -f tmp/root/etc/.pwd.lock
rm -f tmp/root/etc/passwd-
rm -f tmp/root/etc/group-
rm -f tmp/root/etc/shadow-
rm -f tmp/root/etc/gshadow-
clean_chroot

log_status "$0" "SUCCESS"
restore_config
trap '' 0

