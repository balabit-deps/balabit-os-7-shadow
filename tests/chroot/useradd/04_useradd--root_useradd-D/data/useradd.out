GROUP=10
HOME=/tmp
INACTIVE=12
EXPIRE=2007-12-02
SHELL=/bin/sh
SKEL=/etc/skel
CREATE_MAIL_SPOOL=no
